GDJS/Runtime/pixi-renderers/pixi-object-effects-manager.ts

```diff
@@ -0,0 +1,208 @@
+/*
+ * GDevelop JS Platform
+ * Copyright 2013-2021 Florian Rival (Florian.Rival@gmail.com). All rights reserved.
+ * This project is released under the MIT License.
+ */
+namespace gdjs {
+  import PIXI = GlobalPIXIModule.PIXI;
+
+  class PixiObjectEffectsManager {
+    protected _filters: Hashtable<Hashtable<PixiFiltersTools.Filter>>;
+
+    constructor() {
+      this._filters = new Hashtable();
+    }
+
+    update(runtimeObject: RuntimeObject, layer: Layer) {
+      const filters: Hashtable<
+        PixiFiltersTools.Filter
+      > | void = this._filters.get(runtimeObject.getName());
+      if (!filters) return;
+      const filterValues: PixiFiltersTools.Filter[] = [];
+      filters.values(filterValues);
+      for (const filter of filterValues) {
+        filter.update(filter.pixiFilter, layer);
+      }
+    }
+
+    /**
+     * Add a new effect on a runtime object, or replace the one
+     * with the same name.
+     * @param runtimeObject The runtime object
+     * @param effectData The data of the effect to add.
+     */
+    addEffect(
+      runtimeObject: RuntimeObject,
+      effectData: EffectData,
+      layer: Layer
+    ) {
+      const filterCreator = gdjs.PixiFiltersTools.getFilterCreator(
+        effectData.effectType
+      );
+      if (!filterCreator) {
+        console.log(
+          'Filter "' +
+            effectData.name +
+            '" has an unknown effect type: "' +
+            effectData.effectType +
+            '". Was it registered properly? Is the effect type correct?'
+        );
+        return;
+      }
+
+      const filter: PixiFiltersTools.Filter = {
+        pixiFilter: filterCreator.makePIXIFilter(layer, effectData),
+        updateDoubleParameter: filterCreator.updateDoubleParameter,
+        updateStringParameter: filterCreator.updateStringParameter,
+        updateBooleanParameter: filterCreator.updateBooleanParameter,
+        update: filterCreator.update,
+      };
+
+      const renderer = runtimeObject.getRendererObject();
+      renderer.filters = (renderer.filters || []).concat(filter.pixiFilter);
+
+      if (!this._filters.containsKey(runtimeObject.getName())) {
```

Comment:
This approach unfortunately won't really work, because you're storing filters for objects according to their name, which has potential issues if we later want to store more information differing for each instance. Also, it's going to be quite slow, at least suboptimal, to use a hashtable to get the filters everytime we need it for an object, having to read the object name and use it as a key.

Instead, each gdjs.RuntimeObject must have its `_filters`. Much like every layer has it :)

Reply:
It's still fine I think, because we consider `_filters` as an "opaque" data structure which is handled by `gdjs.PixiObjectEffectsManager`. 
In a gdjs.RuntimeObject, there is already usually a `_renderer` object (depending on the object, but it's always and by convention called like this). We now add a `_filters`, to gdjs.RuntimeObject, which has a structure that is both known (it's a map to the current filters applied to the object) but that also contain the "renderer specific filters", aka here the "PIXI filters". 
If we were to implement this for another renderer, like Cocos, we would still have it, but we would have a `gdjs.CocosObjectEffectsManager` that would store cocos effects inside.